### ISSUE TITLE
Custom transformations

### ISSUE THREAD

**Issue by kerkovits** [qgis/QGIS#46382]
### Feature description

Please consider to provide a way to define a custom transformation pipeline between two defined CRS-s. A possibility to manually edit the proj pipeline string in the panel Project->Properties->Transformations. If this custom transformation pipeline is valid according to the PROJ library, the project file would store this manually described string, and use for calculation within the scope of the project.

I imagine it as after one has chosen a transformation for his pair of CRS-s, the cell containing the chosen PROJ pipeline, would be editable. If one modifies it and presses ENTER, QGIS tests if the transformation is valid according to PROJ, and will use that pipeline in the project.

### Additional context

Recent developments in QGIS enhanced the calculations of transformations. This made it possible to use the projection engine of QGIS for high accuracy calculations. Thank you for your great work.

In the present situations the user may choose a transformation between two CRS-s in the panel Project->Properties->Transformations. Clicking on the pencil gives the user a choice between transforms implemented in PROJ. If one would not like to use any of these possibilities but to describe a custom transformation between the two CRS, the database proj.db must be edited. This is a cumbersome work, because a lot of fields must be filled, and at least two tables (e. g. grid_transformation and usage) must be modified. Furthermore, these manual edits are undone by each update of the PROJ library.

The proposal above would mean a more straightforward and persistent way to define custom transformations, which can be used for testing and development purposes. The PROJ pipeline syntax is well documented, so this enhancement could be useful for GIS professionals.

Thank you very much in advance for your consideration. I hope, it is not a large modification on the codebase.

**Comment by gioman** [comment]
@kerkovits 

settings > options > CRS and transforms > transforms 

?

**Comment by kerkovits** [comment]
@gioman I do not fully understand your question. The panel you describe can be used for choosing between transformations **already implemented** by PROJ. It does not allow me to manually type in a custom pipeline. If I would like to test my new pipeline, I must edit proj.db, add the new transform and a new usage record for it. Only after that can I select my custom transform there. However, this approach is a bit too complicated, as described above.

**Comment by gioman** [comment]
> @gioman I do not fully understand your question. The panel you describe can be used for choosing between transformations **already implemented** by PROJ. It does not allow me to manually type in a custom pipeline. If I would like to test my new pipeline, I must edit proj.db, add the new transform and a new usage record for it. Only after that can I select my custom transform there. However, this approach is a bit too complicated, as described above

@kerkovits in 

settings > custom projections

you can create custom CRSs you can use in the other dialog.

**Comment by kerkovits** [comment]
> > @gioman I do not fully understand your question. The panel you describe can be used for choosing between transformations **already implemented** by PROJ. It does not allow me to manually type in a custom pipeline. If I would like to test my new pipeline, I must edit proj.db, add the new transform and a new usage record for it. Only after that can I select my custom transform there. However, this approach is a bit too complicated, as described above
> 
> @kerkovits in
> 
> settings > custom projections
> 
> you can create custom CRSs you can use in the other dialog.

Of course, but in that case I must redefine the CRS of all layers in the project. If I would like to apply a custom transform between two _existing_ systems, this solution is kind of a workaround.

**Comment by nyalldawson** [comment]
Duplicate of https://github.com/qgis/QGIS/issues/31106
